app/uk/gov/hmrc/eusubsidycompliancefrontend/connectors/Connector.scala

```diff
@@ -29,13 +29,14 @@ trait Connector {
   protected val http: HttpClient
 
   protected def makeRequest(
-    request: HttpClient => Future[HttpResponse]
+    request: HttpClient => Future[HttpResponse],
+    isRetrieveEmail: Boolean = false
   )(implicit ec: ExecutionContext): ConnectorResult =
     request(http) map { r: HttpResponse =>
-      if (r.status.isSuccess) Right(r)
+      if (r.status.isSuccess(isRetrieveEmail)) Right(r)
```

Comment:
I think it would be cleaner to introduce an EmailConnector trait that extends Connector and overrides the makeRequest method to provide this behaviour. The Retrieve and Send connectors can then extend this and this new behaviour is then confined to email connectors only.

Reply:
I can do that

Reply:
Thanks!